sdks/java/core/src/main/java/org/apache/beam/sdk/io/CountingSource.java

```diff
@@ -354,7 +356,7 @@ public UnboundedReader<Long> createReader(PipelineOptions options, CounterMark c
 
     @Override
     public Coder<CountingSource.CounterMark> getCheckpointMarkCoder() {
-      return AvroCoder.of(CountingSource.CounterMark.class);
+      return SerializableCoder.of(CountingSource.CounterMark.class);
```

Comment:
I think this needs to change too. (And no need to add the Serializable interface anymore.)

Reply:
Right, my bad. Finally, I decided to create a custom coder for this to keep it compatible with current interfaces (not a case if we change it to `SchemaCoder`).